src/gsi/bkgvar.f90

```diff
@@ -105,15 +105,17 @@ subroutine bkgvar(cvec,iflg)
   call gsi_bundlegetpointer(cvec,'sst',ptrsst,istatus)
   call gsi_bundlegetpointer(cvec,'stl',ptrstl,istatus)
   call gsi_bundlegetpointer(cvec,'sti',ptrsti,istatus)
-  sst=zero
-  stl=zero
-  sti=zero
+  if(iflg == 0)then
+     sst=zero
```

Comment:
Is it necessary to allocate memory for sst at all? Can line 129 be
 ptrsst(j,i)=ptrsst(j,i)*dssvs(j,i,n)
If not, maybe the memory allocation for sst can be moved to within this if statement.

Reply:
I think it is necessary to do as is.  By having the sst array the non-sst points are zeroed out.